dotnet/src/Connectors/Connectors.Memory.Milvus/MilvusMemoryStore.cs

```diff
@@ -0,0 +1,429 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.Memory;
+using Milvus.Client;
+
+namespace Microsoft.SemanticKernel.Connectors.Memory.Milvus;
+
+public class MilvusMemoryStore : IMemoryStore
+{
+    private readonly MilvusClient _milvusClient;
+    private readonly long _vectorSize;
+    private readonly SimilarityMetricType _metricType;
+
+    private const string IsReferenceFieldName = "is_reference";
+    private const string ExternalSourceNameFieldName = "external_source_name";
+    private const string IdFieldName = "id";
+    private const string DescriptionFieldName = "description";
+    private const string TextFieldName = "text";
+    private const string AdditionalMetadataFieldName = "additional_metadata";
+    private const string EmbeddingFieldName = "embedding";
+    private const string KeyFieldName = "key";
+    private const string TimestampFieldName = "timestamp";
+
+    private const ConsistencyLevel DefaultConsistencyLevel = ConsistencyLevel.Session;
+    private const int DefaultVarcharLength = 65_535;
+
+    private readonly QueryParameters _queryParametersWithEmbedding = new()
+    {
+        OutputFields = { IsReferenceFieldName, ExternalSourceNameFieldName, IdFieldName, DescriptionFieldName, TextFieldName, AdditionalMetadataFieldName, EmbeddingFieldName, KeyFieldName }
+    };
+
+    private readonly QueryParameters _queryParametersWithoutEmbedding = new()
+    {
+        OutputFields = { IsReferenceFieldName, ExternalSourceNameFieldName, IdFieldName, DescriptionFieldName, TextFieldName, AdditionalMetadataFieldName, KeyFieldName }
+    };
+
+    private readonly SearchParameters _searchParameters = new()
+    {
+        OutputFields = { IsReferenceFieldName, ExternalSourceNameFieldName, IdFieldName, DescriptionFieldName, TextFieldName, AdditionalMetadataFieldName, KeyFieldName }
+    };
+
+    #region Constructors
+
+    // TODO: Possibly add sugar constructor overloads that create the MilvusClient internally; though that would mean lots and lots of parameters.
```

Comment:
Yeah, absolutely - it's mostly that I discovered some problems with the current constructor overloads in Milvus.Client (ambiguity problems because of optional parameters), need to solve that first.

Reply:
This is now done. There are now three basic constructors:
* One without auth (good for getting started, where Milvus auth isn't enabled)
* One with username/password
* One with an API key (for Zilliz cloud)

Plus one additional advanced constructor, accepting a MilvusClient. I think that's probably the best compromise here in terms of simplicity, without pushing people in silly directions (such as making it complicated to do proper auth).